### PR TITLE
CVSB-185

### PR DESCRIPTION
**Pull request by mdclarke21**
'As a VSA I want to view vehicle details so that I can confirm it is the correct vehicle to conduct testing'

I have updated the following pages: 
- Vehicle details
- Brakes
- Weights
- Tyres
- Additional details

These should now display the dummy data from vehicle-data.mock.ts.

I have also updated the scss files for each page to fix the header text position.

### CODE REVIEW

File: src/pages/testing/test-creation/vehicle/vehicle-details/vehicle-details.ts [hunk 1]
```diff
@@ -65,18 +65,26 @@ export class VehicleDetailsPage {
   }
 
   showVehicleBrakes(): void {
-    this.navCtrl.push('VehicleBrakesPage');
+    this.navCtrl.push('VehicleBrakesPage', {
```
Comment (danieliordache) [inline src/pages/testing/test-creation/vehicle/vehicle-details/vehicle-details.ts]: remove duplicate methods, send page name as a param to a single method

File: src/pages/testing/test-creation/vehicle/vehicle-tyres/vehicle-tyres.html [hunk 2]
```diff
@@ -7,110 +7,70 @@
...
+    <ng-container *ngFor="let vrm of vehicleData.vrms">
+        <div text-uppercase 
+             text-center 
+             class="vehicle-header-vrm-number" 
+             *ngIf="vrm.isPrimary">
+            {{ vrm.vrm }}
+        </div>
+    </ng-container>
+    <ng-container *ngFor="let techRecord of vehicleData.techRecord">
+        <ng-container *ngIf="techRecord.statusCode == 'current'">
```
Comment (danieliordache) [inline src/pages/testing/test-creation/vehicle/vehicle-tyres/vehicle-tyres.html]: move to a method and use toLowerCase() method

File: src/pages/testing/test-creation/vehicle/vehicle-weights/vehicle-weights.html [hunk 3]
```diff
@@ -7,124 +7,102 @@
...
+    <ng-container *ngFor="let vrm of vehicleData.vrms">
+        <div text-uppercase 
+             text-center 
+             class="vehicle-header-vrm-number" 
+             *ngIf="vrm.isPrimary">
+            {{ vrm.vrm }}
+        </div>
+    </ng-container>
+    <ng-container *ngFor="let techRecord of vehicleData.techRecord">
+        <ng-container *ngIf="techRecord.statusCode == 'current'">
```
Comment (danieliordache) [inline src/pages/testing/test-creation/vehicle/vehicle-weights/vehicle-weights.html]: same as the file above

File: src/pages/testing/test-creation/vehicle/vehicle-additional/vehicle-additional.html [hunk 4]
```diff
@@ -7,96 +7,108 @@
...
+    <ng-container *ngFor="let vrm of vehicleData.vrms">
+        <div text-uppercase 
+             text-center 
+             class="vehicle-header-vrm-number" 
+             *ngIf="vrm.isPrimary">
+            {{ vrm.vrm }}
+        </div>
+    </ng-container>
+    <ng-container *ngFor="let techRecord of vehicleData.techRecord">
+        <ng-container *ngIf="techRecord.statusCode == 'current'">
```
[the remark this answers is not in the document]
Comment (danieliordache) [reply src/pages/testing/test-creation/vehicle/vehicle-additional/vehicle-additional.html]: create a method in commonFunctions, import the class as DI for each page of this story. Declare the DI as public and use the method directly into HTML

[the remark this answers is not in the document]
Comment (mdclarke21) [reply src/pages/testing/test-creation/vehicle/vehicle-additional/vehicle-additional.html]: What methods do you want me to create in the commonFunctions?

A function to check the tech record status and a function to check the vehicle type?

[the remark this answers is not in the document]
Comment (mdclarke21) [reply src/pages/testing/test-creation/vehicle/vehicle-additional/vehicle-additional.html]: Done

File: src/pages/testing/test-creation/vehicle/vehicle-details/vehicle-details.html [hunk 5]
```diff
@@ -1,8 +1,8 @@
 <ion-header>
     <ion-navbar>
         <ion-title margin-left>
-            <span *ngIf="vehicle.type != 'Trailer'">Vehicle details</span>
-            <span *ngIf="vehicle.type == 'Trailer'">Trailer details</span>
+            <span *ngIf="vehicleData.vehicleType != 'TRL'">Vehicle details</span>
```
[the remark this answers is not in the document]
Comment (mdclarke21) [reply src/pages/testing/test-creation/vehicle/vehicle-details/vehicle-details.html]: done

File: src/pages/testing/test-creation/vehicle/vehicle-details/vehicle-details.html [hunk 6]
```diff
@@ -13,100 +13,114 @@
...
+            </ng-container>
+        </ng-container>
     </ion-list>
 
     <ion-list class="vehicle-list">
         <ion-list-header padding-left padding-right>VIEW MORE INFORMATION</ion-list-header>
         <button ion-item margin-left>
-            <h3>Vehicle test history</h3>
+            <h3>
+                <span *ngIf="vehicleData.vehicleType != 'TRL'">Vehicle</span>
```
[the remark this answers is not in the document]
Comment (mdclarke21) [reply src/pages/testing/test-creation/vehicle/vehicle-details/vehicle-details.html]: done

File: src/pages/testing/test-creation/vehicle/vehicle-tyres/vehicle-tyres.html (hunk 2 above)
Comment (mdclarke21) [reply src/pages/testing/test-creation/vehicle/vehicle-tyres/vehicle-tyres.html]: done

File: src/pages/testing/test-creation/vehicle/vehicle-weights/vehicle-weights.html (hunk 3 above)
Comment (mdclarke21) [reply src/pages/testing/test-creation/vehicle/vehicle-weights/vehicle-weights.html]: done

File: src/pages/testing/test-creation/vehicle/vehicle-details/vehicle-details.html [hunk 7]
```diff
@@ -13,100 +13,114 @@
...
-            <ion-label>
-                <h3>Brake code</h3>
-            </ion-label>
-            <div item-content padding-right>
-                <p>100A01</p>
+                <p>{{ vehicleData.vin }}</p>
             </div>
         </ion-item>
+        <ng-container *ngFor="let techRecord of vehicleData.techRecord">
+            <ng-container *ngIf="techRecord.statusCode == 'current'">
```
[the remark this answers is not in the document]
Comment (mdclarke21) [reply src/pages/testing/test-creation/vehicle/vehicle-details/vehicle-details.html]: done

File: src/pages/testing/test-creation/vehicle/vehicle-additional/vehicle-additional.html [hunk 8]
```diff
@@ -7,96 +7,108 @@
...
+            
+            <ion-list class="vehicle-list">
+                <ion-list-header>
+                </ion-list-header>
+                <ion-item margin-left>
+                    <ion-label>
+                        <h3>COIF date</h3>
+                    </ion-label>
+                    <div item-content padding-right>
+                        <p>{{ techRecord.coifDate | date: 'dd/MM/yyyy'}}</p>
```
[the remark this answers is not in the document]
Comment (mdclarke21) [reply src/pages/testing/test-creation/vehicle/vehicle-additional/vehicle-additional.html]: Done

File: src/pages/testing/test-creation/vehicle/vehicle-details/vehicle-details.html [hunk 9]
```diff
@@ -13,100 +13,114 @@
...
+                    <div item-content padding-right>
+                        <p>{{ techRecord.manufactureDate }}</p>
+                    </div>
+                </ion-item>
+                <ion-item margin-left>
+                    <ion-label>
+                        <h3>Date of first registration</h3>
+                    </ion-label>
+                    <div item-content padding-right>
+                        <p>{{ techRecord.regnDate | date: 'dd/MM/yyyy'}}</p>
```
[the remark this answers is not in the document]
Comment (mdclarke21) [reply src/pages/testing/test-creation/vehicle/vehicle-details/vehicle-details.html]: Done

File: src/providers/utils/common-functions.ts [hunk 10]
```diff
@@ -3,4 +3,9 @@ export class CommonFunctionsService {
   public capitalizeString(string: string) {
     return string.charAt(0).toUpperCase() + string.slice(1);
   }
-}
+
+  public checkForMatch(inputValue: string, expectedValue: string){
+    return inputValue.toLowerCase() === expectedValue.toLowerCase() ? true : false;
```
Comment (danieliordache) [inline src/providers/utils/common-functions.ts]: you can remove " ? true : false", it's redundant.

**Pull request merged**